CMakeLists.txt

```diff
@@ -1,47 +1,109 @@
+else ()
+    message (STATUS "${Green}Build type is: ${CMAKE_BUILD_TYPE}${ColourReset}")
+endif()
+if(POLICY CMP0135)
+    cmake_policy(SET CMP0135 NEW)
+endif()
 if (${CMAKE_BUILD_TYPE} STREQUAL "Release")
     add_definitions (-DNOPAssert)
 endif ()
 
 add_compile_options (-Wall)
 add_compile_options (-Wunused)
 add_compile_options (-Wextra)
-add_compile_options (-Werror)
+#add_compile_options (-Werror)
 
 # allow deprecated functions
 add_compile_options (-Wno-deprecated-declarations)
 
 option (USE_STATIC_LIBRARIES "Link with static libraries if available" ON)
 
-message (STATUS "Build type is: " ${CMAKE_BUILD_TYPE})
+if(NOT IPPL_PLATFORMS)
+    message(STATUS "${Yellow}IPPL_PLATFORMS flag is not set, defaulting to serial backend${ColourReset}")
+    set(IPPL_PLATFORMS "SERIAL")
+endif()
 
-find_package(Kokkos 4.0.0 REQUIRED)
+if(IPPL_PLATFORMS)
+    #message(STATUS "IPPL_PLATFORMS flag is set to: ${IPPL_PLATFORMS}")
+    string(TOUPPER "${IPPL_PLATFORMS}" IPPL_PLATFORMS)
+    # Perform case distinction
+    if("${IPPL_PLATFORMS}" STREQUAL "CUDA")
+        message(STATUS "${Green}CUDA backend enabled${ColourReset}")
+        set(Kokkos_ENABLE_CUDA ON CACHE BOOL "Enable Kokkos CUDA Backend")
+        set(Heffte_ENABLE_CUDA ON CACHE BOOL "Set Heffte CUDA Backend")
+    elseif("${IPPL_PLATFORMS}" STREQUAL "OPENMP")
+        message(STATUS "${Green}OpenMP backend enabled${ColourReset}")
+        if(NOT Heffte_ENABLE_FFTW)
+            message(STATUS "${Yellow}Because Heffte_ENABLE_FFTW is not set defaulting to Stock with AVX${ColourReset}")
+            set(Heffte_ENABLE_AVX2 ON CACHE BOOL "Enable Heffte Stock backend lol")
+        else()
+            message(STATUS "${Green}Enabled FFTW${ColourReset}")
+        endif()
+        set(Kokkos_ENABLE_OPENMP ON CACHE BOOL "Enable Kokkos OpenMP Backend")
+    elseif("${IPPL_PLATFORMS}" STREQUAL "SERIAL")
+
+        message(STATUS "${Green}Serial backend enabled${ColourReset}")
+        if(NOT Heffte_ENABLE_FFTW)
+            message(STATUS "${Yellow}Because Heffte_ENABLE_FFTW is not set defaulting to Stock with AVX${ColourReset}")
+            set(Heffte_ENABLE_AVX2 ON CACHE BOOL "Enable Heffte Stock backend lol")
+        endif()
+
+    elseif("${IPPL_PLATFORMS}" STREQUAL "CUDA;OPENMP" OR "${IPPL_PLATFORMS}" STREQUAL "OPENMP;CUDA")
+        message(STATUS "${Green}CUDA and OpenMP backend enabled${ColourReset}")
+        set(Heffte_ENABLE_CUDA ON CACHE BOOL "Set Heffte CUDA Backend")
+        set(Kokkos_ENABLE_OPENMP ON CACHE BOOL "Enable Kokkos OpenMP Backend")
+        set(Kokkos_ENABLE_CUDA ON CACHE BOOL "Enable Kokkos CUDA Backend") 
+    else()
+        message(FATAL_ERROR "Unknown or unsupported platform: ${IPPL_PLATFORMS}")
+        # Handle unknown or unsupported platform
+    endif()
+endif()
 
+find_package(Kokkos 4.0.0 QUIET)
+if(NOT Kokkos_FOUND)
+    FetchContent_Declare(
+        #DOWNLOAD_EXTRACT_TIMESTAMP
+        kokko
```

Comment:
Typo.

Reply:
I actually named it that on purpose to avoid name duplication (fetchtcontent object vs. target name), but I think that's superfluous.

Reply:
Maybe add a comment and then resolve this.

Reply:
changed it